app/src/main/java/protect/card_locker/ScanActivity.kt

```diff
@@ -1,541 +1,548 @@
-        return super.onOptionsItemSelected(item);
+        return super.onOptionsItemSelected(item)
     }
 
-    private void setScannerActive(boolean isActive) {
+    private fun setScannerActive(isActive: Boolean) {
         if (isActive) {
-            barcodeScannerView.resume();
+            barcodeScannerView.resume()
         } else {
-            barcodeScannerView.pause();
+            barcodeScannerView.pause()
         }
-        mScannerActive = isActive;
+        mScannerActive = isActive
     }
 
-    private void returnResult(ParseResult parseResult) {
-        Intent result = new Intent();
-        Bundle bundle = parseResult.toLoyaltyCardBundle(ScanActivity.this);
-        if (addGroup != null) {
-            bundle.putString(LoyaltyCardEditActivity.BUNDLE_ADDGROUP, addGroup);
+    private fun returnResult(parseResult: ParseResult) {
+        val bundle = parseResult.toLoyaltyCardBundle(this).apply {
+            addGroup?.let { putString(LoyaltyCardEditActivity.BUNDLE_ADDGROUP, it) }
         }
-        result.putExtras(bundle);
-        ScanActivity.this.setResult(RESULT_OK, result);
-        finish();
+        val result = Intent().apply { putExtras(bundle) }
+        this.setResult(RESULT_OK, result)
+        finish()
     }
 
-    private void handleActivityResult(int requestCode, int resultCode, Intent intent) {
-        super.onActivityResult(requestCode, resultCode, intent);
+    private fun handleActivityResult(requestCode: Int, resultCode: Int, intent: Intent?)  {
+        super.onActivityResult(resultCode, resultCode, intent)
 
-        List<ParseResult> parseResultList = Utils.parseSetBarcodeActivityResult(requestCode, resultCode, intent, this);
+        val parseResultList: List<ParseResult> = Utils.parseSetBarcodeActivityResult(requestCode, resultCode, intent, this)
 
         if (parseResultList.isEmpty()) {
-            setScannerActive(true);
-            return;
+            setScannerActive(true)
+            return
         }
 
-        Utils.makeUserChooseParseResultFromList(this, parseResultList, new ParseResultListDisambiguatorCallback() {
-            @Override
-            public void onUserChoseParseResult(ParseResult parseResult) {
-                returnResult(parseResult);
-            }
 
-            @Override
-            public void onUserDismissedSelector() {
-                setScannerActive(true);
+        Utils.makeUserChooseParseResultFromList(
+                this,
+                parseResultList,
+                object : ParseResultListDisambiguatorCallback {
+            override fun onUserChoseParseResult(parseResult: ParseResult) {
+                returnResult(parseResult)
+            }
+            override fun onUserDismissedSelector() {
+                setScannerActive(true)
             }
-        });
+        })
     }
 
-    private void addWithoutBarcode() {
-        AlertDialog.Builder builder = new MaterialAlertDialogBuilder(this);
+    private fun addWithoutBarcode() {
```

Comment:
I feel this function could use some more `.apply` but I don't blame you for not wanting to deal with that. We can skip that for this PR and clean that up further later :)